### PR TITLE
Fix boost 1.86.0 recipe for x86 arch

### PR DESCRIPTION
**Pull request by adaviding**
### Summary
Changes to recipe:  **boost/1.86.0**

### Motivation
Fix https://github.com/conan-io/conan-center-index/issues/25864.

### Details
We can see the relevant part of the build definition for boost 1.86.0 [here](https://github.com/boostorg/stacktrace/blob/boost-1.86.0/build/Jamfile.v2#L148).  The token `$(arch:G=)` will evaluate to `x86` in two conan recipe scenarios:
* settings.arch = "x86"
* settings.arch = "x86_64"

We also see an existing helper property `_b2_architecture` in the conan recipe [here](https://github.com/conan-io/conan-center-index/blob/f7a9ac9570e8f7d6fe806e1ca9b9be5dbdf828a5/recipes/boost/all/conanfile.py#L1206).

Therefore this change seems reasonable.

#### profile "win32_v192_debug.txt"
VS 2019 16.11.41
```
[settings]
arch=x86
build_type=Debug
compiler=msvc
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=192
os=Windows
```
#### profile "x64_v192_debug.txt"
VS 2019 16.11.41
```
[settings]
arch=x86_64
build_type=Debug
compiler=msvc
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=192
os=Windows
```

#### profile "win32_v194_debug.txt"
VS 2022 17.11.5
```
[settings]
arch=x86
build_type=Debug
compiler=msvc
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=194
os=Windows
```
#### profile "x64_v194_debug.txt"
VS 2022 17.11.5
```
[settings]
arch=x86_64
build_type=Debug
compiler=msvc
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=194
os=Windows
```

#### cases
Using Conan 2.9.2:
* [x] `conan create ./all --version 1.86.0 --build missing --profile:all "C:\tmp\conan_profiles\win32_v192_debug.txt"`
* [ ] `conan create ./all --version 1.86.0 --build missing --profile:all "C:\tmp\conan_profiles\x64_v192_debug.txt"`
* [x] `conan create ./all --version 1.86.0 --build missing --profile:all "C:\tmp\conan_profiles\win32_v194_debug.txt"`
* [x] `conan create ./all --version 1.86.0 --build missing --profile:all "C:\tmp\conan_profiles\x64_v194_debug.txt"`
---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by uilianries — APPROVED** [review]
LGTM. B2 x86 covers both x86 and x86_64: https://www.boost.org/doc/libs/1_86_0/tools/build/doc/html/index.html#bbv2.overview.builtins.features

**Comment by jcar87** [comment]
Thanks @adaviding  - please be aware that as of this week we have stopped supporting Conan 1.x in Conan Center - so while the fix is correct, it will only be made available for Conan 2.x onwards.

**Comment by adaviding** [comment]
@pya102 @robomics @SpaceIm @tpaustian I need one more review for this PR.  Looking at the commit history for boost, you are the last 4 (besides Uilian) to touch it.  I just need 1 of you.

BTW -- GitHub does not allow me to request a review in the normal way.  (It appears to be a permission issue.)

**Comment by robomics** [comment]
@adaviding I believe you need one more approval from someone with write access to this repo.
Unfortunately I am a regular user, so I cannot help you.

**Review by jcar87 — APPROVED** [review]

**Pull request merged**
